src/main/java/com/amazon/opendistro/elasticsearch/performanceanalyzer/config/PluginSettings.java

```diff
@@ -244,16 +244,14 @@ private static void validateOrCreateDir(String path) throws ConfigFatalException
       LOG.info("Trying to create directory {}.", path);
     }
 
-    boolean valid = dictCreated && dict.isDirectory() && dict.canWrite();
+    boolean valid = dictCreated && dict.isDirectory();
```

Comment:
The `PluginSettings` class is statically initialized. So this check happens in the context of the writer and the reader. I agree that writable check is not required for the reader but this is a required check for the writer. So, if we want to remove this check, we have to selectively do this only for the reader.